Tests/XcodeProjTests/Project/XcodeProjTests.swift

```diff
@@ -25,6 +25,19 @@ final class XcodeProjIntegrationTests: XCTestCase {
                                         initModel: XcodeProj.init(path:))
     }
     
+    func test_initialize_PBXProj_with_data() {
+        let pbxprojPath = iosProjectPath.url.appendingPathComponent("project.pbxproj")
+        
+        if let pbxprojStringRepresentation = try? String(contentsOf: pbxprojPath) {
+            let pbxprojData = Data(pbxprojStringRepresentation.utf8)
+            
+            XCTAssertNotNil(try? PBXProj(data: pbxprojData))
+            return
+        }
+
+        XCTFail()
+    }
```

Comment:
For completeness it would be good to check the correctness of initialing from data representation in addition to verifying it's not `nil`

```suggestion
    func test_initialize_PBXProj_with_data() throws {
        // Given
        let pbxprojPath = iosProjectPath + "project.pbxproj"
        let pbxprojFromDisk = try PBXProj(path: pbxprojPath)
        let pbxprojData = try Data(contentsOf: pbxprojPath.url)

        // When
        let pbxprojFromData = try PBXProj(data: pbxprojData)
        try pbxprojFromData.updateProjectName(path: pbxprojPath)

        // Then
        XCTAssertEqual(pbxprojFromData, pbxprojFromDisk)
    }
```

actually doing so reveals initialising from data representation may require also specifying the `name` which currently doesn't get serialised (from a cursory glance, doesn't seem to be part of the project format) .